server/src/main/java/io/crate/planner/CreateUserMappingPlan.java

```diff
@@ -87,8 +88,9 @@ public void executeOrFail(DependencyCarrier dependencies,
             plannerContext.nodeContext(),
             subQueryResults
         ).bind(params);
-        Map<String, Object> options = createUserMapping.options().entrySet().stream()
-            .collect(Collectors.toMap(Entry::getKey, entry -> toValue.apply(entry.getValue())));
+        Map<String, String> optionsMap = createUserMapping.options().entrySet().stream()
+            .collect(Collectors.toMap(Entry::getKey, entry -> toValue.apply(entry.getValue()).toString()));
+        var options = Settings.builder().putProperties(optionsMap, s -> s).build();
```

Comment:
This could break for list options if the value is converted via `.toString()` and creating a map just to then transfer it into another map in the settings builder is a waste.

Could do this like in https://github.com/crate/crate/blob/0f47b8185b6681f06f3798c5392d11646a377d46/server/src/main/java/io/crate/planner/CreateServerPlan.java#L74-L90

Reply:
Thanks updated